kubernetes-1.25.yaml

```diff
@@ -0,0 +1,145 @@
+package:
+  name: kubernetes-1.25
+  version: 1.25.11
+  epoch: 0
+  description: Production-Grade Container Scheduling and Management
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - bash
+      - ca-certificates-bundle
+      - build-base
+      - go
+      - go-bindata
+      - linux-headers
+      - grep
+      - jq
+      - coreutils # needed for non busybox version of `mktemp`
+      - findutils # needed for non busybox version of `xargs`
+      - rsync
+      - apk-tools
+
+# "transform" the kubernetes version into the corresponding pause version, these don't always line up
+var-transforms:
+  - from: ${{package.version}}
+    match: ^(.+)$
+    replace: 3.9
+    to: pause-version
+
+vars:
+  components: "kubectl kubeadm kubelet kube-scheduler kube-proxy kube-controller-manager kube-apiserver"
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/kubernetes/kubernetes
+      tag: v${{package.version}}
+      expected-commit: 8cfcba0b15c343a8dc48567a74c29ec4844e0b9e
+
+  - runs: |
+      ./hack/pin-dependency.sh github.com/docker/distribution v2.8.2
+      ./hack/update-vendor.sh
+      ./hack/lint-dependencies.sh
+
+  - runs: |
+      WHAT=""
+      for c in ${{vars.components}} ; do
+        WHAT="$WHAT cmd/$c"
+      done
+
+      make WHAT="$WHAT"
+
+  - runs: |
+      mkdir -p "${{targets.destdir}}"/usr/bin/
+      install -m755 _output/bin/* ${{targets.destdir}}/usr/bin/
+
+      mkdir -p "${{targets.destdir}}"/etc/kubernetes
+
+subpackages:
+  - name: kubectl
```

Comment:
I think we want each of these to have a `provides: ['kubectl=1.25']` so that folks downstream don't have to care about the version to find the package.

Reply:
thanks, just caught this :D

Reply:
actually... do we edo this with a `provides: ['kubectl=1.15']` or with just renaming the subpackage with the variant? `name: kubectl-1.25`

Reply:
I went with the approach we use for `python` (both)